Redux Toolkit 🛠🚀

jakiestfu opened the pull request:
Convert all files to use redux for game state and subsequently remove the gameState file

- [x] Emoji.ts
- [x] attack/
- [x] character/
- [x] commands/
    - [x] admin.ts
  - [x] barFight.ts
  - [x] cooldowns.ts
  - [x] db.ts
  - [x] hp.ts
  - [x] inventory.ts
  - [x] lootme.ts
  - [x] primaryStatFields.ts
  - [x] quests.ts
  - [x] set.ts
  - [x] adventure.ts
  - [x] buyItem.ts
  - [x] crown.ts
  - [x] equip.ts
  - [x] index.ts
  - [x] inventorySelector.ts
  - [x] lootmonster.ts
  - [x] quest.ts
  - [x] randomInventoryItem.ts
  - [x] shop.ts
  - [x] attack.ts
  - [x] chest.ts
  - [x] dance.ts
  - [x] heal.ts
  - [x] inspect.ts
  - [x] list
  - [x] monster.ts
  - [x] questEmbed.ts
  - [x] renew.ts
  - [x] statusEffectEmbed.ts
- [x] encounter/
- [x] encounters/
- [x] equipment/
- [x] gameState.ts
- [x] getCooldownRemaining.ts
- [x] grantDivineBlessing.ts
- [x] heal/
- [x] heavyCrown/
- [x] index.ts
- [x] monster/
- [x] quest/
- [x] statusEffects/
- [x] store/
- [x] trap/
- [x] utils/
- [x] utils.ts

jakiestfu committed:
initial commit

jakiestfu committed:
typo

jakiestfu committed:
more reducers

jakiestfu committed:
misc

jakiestfu committed:
more additions

jakiestfu committed:
paths and more

jakiestfu committed:
move build folder

jakiestfu committed:
more conversions

jakiestfu committed:
moar

brian-gates committed:
getCharacter(Update) updates

brian-gates committed:
updateMonster

brian-gates committed:
character user not required

jakiestfu committed:
misc updates

jakiestfu committed:
undo ts pathing

brian-gates committed:
Merge branch 'main' into feat/redux-toolkit

brian-gates committed:
wip

jakiestfu committed:
fix devtools

brian-gates committed:
getMonster from redux

brian-gates committed:
fixed attack results

brian-gates committed:
fix typo

brian-gates committed:
fixed loot

brian-gates committed:
Merge branch 'main' into feat/redux-toolkit

brian-gates committed:
fix character slice

brian-gates committed:
fix chest

brian-gates committed:
but really this time

brian-gates committed:
it lives again!

brian-gates committed:
this seems to screw with types

brian-gates committed:
redux devtool launcher

brian-gates committed:
moar fixes

brian-gates committed:
move getMonsterUpdate under monster

brian-gates committed:
move getMonster

brian-gates committed:
cleanup

brian-gates committed:
touch of color

brian-gates committed:
refactors

brian-gates committed:
refactor

brian-gates committed:
fix getRandomMonster

brian-gates committed:
getRoamingMonsters and deleting gameState

Co-authored-by: Jacob Kelley <jakiestfu@users.noreply.github.com>

jakiestfu committed:
persist redux

brian-gates commented:
Bug - monsters result in 0xp, sometimes 0gp.
![image](https://user-images.githubusercontent.com/97096/144124331-6e725c8c-0f02-41f6-8d4f-9073bd1f5886.png)

jakiestfu committed:
This is a combination of 2 commits.

ensure relative pathing is used for auto import

ensure relative pathing is used for auto import

jakiestfu committed:
kill gameState

brian-gates commented on src/character/getCharacterUpdate.ts:
```diff
@@ -1,13 +1,10 @@
+import store from "../store";
+import { getCharacterById } from "../store/selectors";
 
 export const getCharacterUpdate = (character: Character): Character => {
```
We might be able to kill this function now.

brian-gates commented on src/commands/db.ts:
```diff
@@ -23,24 +24,25 @@ subcommands.set("save", async (interaction: CommandInteraction) => {
-  } catch (e) {
-    await interaction.reply(`Database load FAILED. ${e}`);
-  }
-});
+// subcommands.set("load", async (interaction: CommandInteraction) => {
+//   if (!interaction.memberPermissions?.has(Permissions.FLAGS.ADMINISTRATOR)) {
+//     return await interaction.reply("Admin required.");
+//   }
+//   try {
+//     loadDB();
+//     await interaction.reply("Database loaded successfully.");
+//   } catch (e) {
+//     console.log(e)
+//     await interaction.reply(`Database load FAILED. ${e}`);
+//   }
+// });
```
I think we should just delete this.

brian-gates replied:
I deleted this.

brian-gates committed:
fix build, consolidate DB_FILE

brian-gates committed:
cleaning up comments and dead code

brian-gates committed:
Merge branch 'main' into feat/redux-toolkit
